stripe.go

```diff
@@ -929,16 +929,12 @@ func (s *BackendImplementation) shouldRetry(err error, req *http.Request, resp *
 		}
 	}
 
-	// 500 Internal Server Error
+	// Retry on 500, 503, and other internal errors.
 	//
-	// We only bother retrying these for non-POST requests. POSTs end up being
-	// cached by the idempotency layer so there's no purpose in retrying them.
-	if resp.StatusCode >= http.StatusInternalServerError && req.Method != http.MethodPost {
-		return true, ""
-	}
-
-	// 503 Service Unavailable
-	if resp.StatusCode == http.StatusServiceUnavailable {
+	// Note that we expect the stripe-should-retry header to be false
+	// in most cases when a 500 is returned, since our idempotency framework
+	// would typically replay it anyway.
+	if resp.StatusCode >= http.StatusInternalServerError {
```

Comment:
Nit:
```suggestion
	if resp.StatusCode >= 500 {
```
Maybe you don't feel the same way, but I find `>=` super awkward with the constants. What matters is that the status is 500-level, not that the status is greater than the specific status that happens to mean "internal server error"

Reply:
I don't feel strongly about this either way - so am going to take the path of least resistance which is maintain status quo